src/shared/components/query/QueryStore.ts

```diff
@@ -563,6 +499,181 @@ export class QueryStore
+		await: ()=>[this.physicalStudiesSet, this.userVirtualStudiesSet],
+		invoke: async ()=>{
+			let physicalStudiesIdsSet = this.physicalStudiesSet.result
+			let virtualStudiesIdsSet = this.userVirtualStudiesSet.result
+
+			let knownSelectableIds = Object.assign([], Object.keys(physicalStudiesIdsSet), Object.keys(virtualStudiesIdsSet));
+
+			//queried id that are not selectable(this would mostly be shared virtual study)
+			const unknownQueriedIds:string[] = this._defaultSelectedIds.keys().filter(id => !_.includes(knownSelectableIds,id));
+
+			let result:{[studyId:string]:VirtualStudy} = {}
+			
+			await Promise.all(unknownQueriedIds.map(id =>{
+				return new Promise((resolve, reject) => {
+					sessionServiceClient.getVirtualStudy(id).then((virtualStudy)=>{
+						//physical study ids iin virtual study
+						let ids = virtualStudy.data.studies.map(study=>study.id);
+						//unknown/unauthorized studies within virtual study
+						let unKnownPhysicalStudyIds = ids.filter(id => !physicalStudiesIdsSet[id]);
+						if(_.isEmpty(unKnownPhysicalStudyIds)){
+							result[id] = virtualStudy;
+						}
+						resolve();
+					}).catch(() => {//error is thrown when the id is not found
+						resolve();
+					});
+				});
+			}));
+			return result;
+		},
+		default: {}
+	});
+
+	//this is mainly used when custom case set is selected
+	private readonly selectedStudyToSampleSet = remoteData<{[id:string]:{[id:string]:boolean}}>({
+		await: ()=>[this.userVirtualStudiesSet, this.sharedVirtualStudiesSet],
+		invoke: async ()=>{
+			let studyToSampleSet:{[id:string]:{[id:string]:boolean}} = {};
+
+			const _vs = {...this.userVirtualStudiesSet.result, ...this.sharedVirtualStudiesSet.result};
+
+			const physicalStudyIds = _.filter(this._allSelectedStudyIds.keys(),studyId=> _vs[studyId] === undefined);
+
+			if(this._allSelectedStudyIds.keys().length !== physicalStudyIds.length){
+				await Promise.all(_.map(physicalStudyIds, studyId => {
+					return client.getAllSamplesInStudyUsingGET({
+						studyId: studyId
+					}).then(samples => {
+						studyToSampleSet[studyId] = stringListToSet(samples.map(sample => sample.sampleId));
+					})
+				}));
+	
+				this._allSelectedStudyIds.keys().forEach(id=>{
+					if(_vs[id]){
+						let virtualStudy = _vs[id];
+						virtualStudy.data.studies.forEach(study => {
+							if(studyToSampleSet[study.id] === undefined){
+								studyToSampleSet[study.id] = stringListToSet(study.samples)
+							}
+						});
+					}
+				});
+			} else {
+				physicalStudyIds.forEach(studyId=>{
+					studyToSampleSet[studyId] = {}
+				})
+			}
+			return studyToSampleSet;
+		},
+		default: {}
+	})
+
+	private readonly selectableStudiesSet = remoteData<{[studyId:string]:string[]}>({
+		await: ()=>[this.physicalStudiesSet, this.selectedStudyToSampleSet],
+		invoke: async ()=>{
+			let result:{[studyId:string]:string[]} = {}
+		result = _.reduce(this.physicalStudiesSet.result,(acc, next, key)=>{

```

Comment:
this is a case where it's confusing to use reduce because you have to pass in a previously assigned variable.   here it's better to use each.   you don't have to change this i'm just pointing it out.